src/components/lib/cos_kernel_api.c

```diff
@@ -421,7 +422,7 @@ cos_captbl_alloc(struct cos_compinfo *ci)
 	assert(ci);
 
 	if (__alloc_mem_cap(ci, CAP_CAPTBL, &kmem, &cap)) return 0;
-	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, ci->pgtbl_cap, kmem, 0)) BUG();
+	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, __compinfo_metacap(ci)->local_mi.pgtbl_cap, kmem, 0)) BUG();
```

Comment:
Yes,  @phani can help us to make sure when we use ci or meta.
I think a general rule is whenever you provide a memory capability (kmem here), you need to provide the corresponding page table who holds that capability.
I think in current code, all memory are allocated from the metta's page table.